mne_sandbox/connectivity/cfc.py

```diff
@@ -121,17 +121,129 @@ def phase_amplitude_coupling(inst, f_phase, f_amp, ixs, pac_func='ozkurt',
+        between n_ch_pairs of channels. Indices correspond to rows of `data`.
+    sfreq : float
+        The sampling frequency of the data.
+    events : array, shape (n_events, 3) | array, shape (n_events,) | None
+        MNE events array. To be supplied if data is 2D and output should be
+        split by events. In this case, `tmin` and `tmax` must be provided. If
+        `ndim == 1`, it is assumed to be event indices, and all events will be
+        grouped together.
+    times : array, shape (n_times,) | None
+        An array of times corresponding to the final axis of inputs. Will be
+        used in conjunction with tmin/tmax if `events` is None.
+    pac_func : {'plv', 'glm', 'mi_canolty', 'mi_tort', 'ozkurt'}
+        The function for estimating PAC. Corresponds to functions in
+        `pacpy.pac`. Defaults to 'ozkurt'. If multiple frequency bands are used
+        then `plv` cannot be calculated.
+    tmin : float | list of floats, shape (n_pac_windows,) | None
+        If `events` is not provided, it is the start time to use in `inst`.
+        If `events` is provided, it is the time (in seconds) to include before
+        each event index. If a list of floats is given, then PAC is calculated
+        for each pair of `tmin` and `tmax`. Defaults to `min(inst.times)`.
+    tmax : float | list of floats, shape (n_pac_windows,) | None
+        If `events` is not provided, it is the stop time to use in `inst`.
+        If `events` is provided, it is the time (in seconds) to include after
+        each event index. If a list of floats is given, then PAC is calculated
+        for each pair of `tmin` and `tmax`. Defaults to `max(inst.n_times)`.
+    n_cycles_ph : float, int | array of floats, shape (n_bands_phase,)
+        The number of cycles to be included in the window for each band-pass
+        filter for phase. Defaults to 3.
+    n_cycles_am : float, int | array of floats, shape (n_bands_amp,)
+        The number of cycles to be included in the window for each band-pass
+        filter for amplitude. Defaults to 3.
+    scale_amp_func : None | function
+        If not None, will be called on each amplitude signal in order to scale
+        the values. Function must accept an N-D input and will operate on the
+        last dimension. E.g., `sklearn.preprocessing.scale`.
+        Defaults to no scaling.
+    n_jobs : int
+        Number of jobs to run in parallel. Defaults to 1.
+    """
+    def __init__(self, f_phase, f_amp, ixs, sfreq, events=None, times=None,
+                 pac_func='ozkurt', tmin=0., tmax=1., n_cycles_ph=3,
+                 n_cycles_am=3, scale_amp_func=None, n_jobs=1):
+        if self.events is not None and self.times is not None:
+            raise ValueError('Must supply one of events and times, not both')
+        self.f_phase = f_phase
+        self.f_amp = f_amp
+        self.ixs = ixs
+        self.sfreq = sfreq
+        self.pac_func = pac_func
+        self.events = events
```

Comment:
Yes, it's a sklearn corner case ATM. But putting it in init isn't the answer, because transformer should be applicable to different dataset without reinit.

I think we should not implement the raw version for now; we'll wait until sklearn solves the possibility to pass more than X and y in their pipeline.